platform/platform.cpp

```diff
@@ -33,6 +42,38 @@ Platform::EError Platform::ErrnoToError()
   }
 }
 
+// static
+bool Platform::RmDirRecursively(string const & dirName)
+{
+  if (dirName.empty() || IsSpecialDirName(dirName))
+    return false;
+
+  bool res = true;
+
+  TFilesWithType fwts;
+  GetFilesByType(dirName, FILE_TYPE_REGULAR | FILE_TYPE_DIRECTORY, fwts);
```

Comment:
Probably you need to get all files in a directory, including fifos, links and other stuff.